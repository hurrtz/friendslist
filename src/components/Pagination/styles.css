:local(.root) {
  display: flex;
  justify-content: space-between;
  padding: 6px 12px;
}
