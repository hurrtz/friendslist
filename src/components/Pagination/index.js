import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import styles from './styles.css';

class Pagination extends PureComponent {
  static props = {
    goBack: PropTypes.func,
    goNext: PropTypes.func,
  };

  renderBackButton() {
    const { goBack } = this.props;

    if (goBack) {
      return (
        <button
          type="button"
          onClick={goBack}
          className={classnames('btn', 'bnt-default')}
        >
          Back
        </button>
      );
    }

    return <div />;
  }

  renderNextButton() {
    const { goNext } = this.props;

    if (goNext) {
      return (
        <button
          type="button"
          onClick={goNext}
          className={classnames('btn', 'bnt-default')}
        >
          Next
        </button>
      );
    }

    return <div />;
  }

  render() {
    const { goBack, goNext } = this.props;

    if (!goBack && !goNext) {
      return null;
    }

    return (
      <div className={styles.root}>
        {this.renderBackButton()}
        {this.renderNextButton()}
      </div>
    );
  }
}

export default Pagination;
