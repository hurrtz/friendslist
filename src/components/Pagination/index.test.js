import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer';

import Pagination from './';

Enzyme.configure({ adapter: new Adapter() });

const defaultProps = {
  goBack: () => {},
  goNext: () => {},
};

function setup(injectedProps = {}) {
  const props = {
    ...defaultProps,
    ...injectedProps,
  };

  const enzymeWrapper = shallow(<Pagination {...props} />);

  return {
    props,
    enzymeWrapper,
  };
}

describe('Pagination', () => {
  it('should render self and subcomponents', () => {
    const { enzymeWrapper, props } = setup();

    expect(enzymeWrapper.find('button').exists()).toBe(true);
  });

  it('should match the snapshot', () => {
    const tree = renderer
      .create(<Pagination goBack={() => {}} goNext={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
