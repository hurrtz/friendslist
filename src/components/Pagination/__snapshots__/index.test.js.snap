exports[`Pagination should match the snapshot 1`] = `
<div
  className={undefined}>
  <button
    className="btn bnt-default"
    onClick={[Function]}
    type="button">
    Back
  </button>
  <button
    className="btn bnt-default"
    onClick={[Function]}
    type="button">
    Next
  </button>
</div>
`;
